dm/master/scheduler/scheduler.go

```diff
@@ -326,6 +326,36 @@ func (s *Scheduler) AddSourceCfg(cfg *config.SourceConfig) error {
 	return nil
 }
 
+// UpdateSourceCfg update the upstream source config to the cluster.
+// please verify the config before call this.
+func (s *Scheduler) UpdateSourceCfg(cfg *config.SourceConfig) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if !s.started {
+		return terror.ErrSchedulerNotStarted.Generate()
+	}
+
+	// 1. check whether the config exists.
+	_, ok := s.sourceCfgs[cfg.SourceID]
+	if !ok {
+		return terror.ErrSchedulerSourceCfgNotExist.Generate(cfg.SourceID)
+	}
+	// 2. check if tasks using this configuration are running
+	if tasks := s.GetTaskNameListBySourceName(cfg.SourceID); len(tasks) > 0 {
+		return terror.ErrSchedulerSourceOpTaskExist.Generate(cfg.SourceID, tasks)
+	}
+	// 3. put the config into etcd.
+	_, err := ha.PutSourceCfg(s.etcdCli, cfg)
```

Comment:
seems the worker will not watch source config, so worker is unaware of source config updated. I prefer we let `UpdateSourceCfg` to return an unimeplemented error for now

Reply:
mybad, forgot to copy update source cfg part from old api codebase 😂 fixed in https://github.com/pingcap/dm/pull/2083/commits/5a104d05872654cf64785b49cfa1cfa99736c644